source/ConversationPanel.cpp

```diff
@@ -454,3 +484,13 @@ Point ConversationPanel::Paragraph::Draw(Point point, const Color &color) const
 	point.Y() += wrap.Height();
 	return point;
 }
+
+
+
+int ConversationPanel::MapChoice(int n) const
+{
+	if(n < 0 || n >= static_cast<int>(choices.size()))
+		return 0;
+	else
+		return next(choices.cbegin(), n)->second;
+}
```

Comment:
If `Conversation::NextNode` should _always_ be called with the output of `MapChoice`, make `MapChoice` return a custom class that wraps an int (aka a "strong type").

That will help prevent new calls from not integrating with the conditional display system.

Reply:
Good idea, though now I'll have to remember how to make a "newtype" wrapper in C++. (Rust has really spoiled me.)

`MappedChoice` seem like an appropriate name? Or, maybe, `ChoiceResult`?

Reply:
On deeper inspection, I like it better as a bare integer. The index given to `Conversation::NextNode` is among all choices present in the data, while the one given to `ConversationPanel::MapChoice` is only among choices *currently* visible in the panel. Other consumers of `NextNode` (of which there are currently only one) don't necessarily want to work with it in a post-conditioned state, and if they do, what shape that state takes should be up to them.